File: IV/Programming in computer networks/Laba1/src/Clothes.java
public class Clothes extends Product {
    private String country;
    private String material;

    Clothes() {
        super();
        country = null;
        material = null;
    }

    Clothes(String n, double p, String c, String m) {
        super(n, p);
        country = c;
        material = m;
    }

    void setCountry(String c) {
        country = c;
    }

    String getCountry() {
        return country;
    }

    void setMaterial(String m) {
        material = m;
    }

    String getMaterial() {
        return material;
    }

    public String toString() {
        return super.toString() + ", Страна производитель: " + country + ", Материал: " + material;
    }

}

File: II/C#/laba8/laba3/Clothes.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace laba3
{
    public class Clothes : Product,IName<Clothes>
    {
        protected string material;
        protected string countryOfOrigin;

        public Clothes()
        {
            material = " ";
            countryOfOrigin = " ";
        }
        public Clothes(string name, double price, string material, string countryOfOrigin) : base(name, price)
        {
            this.material = material;
            this.countryOfOrigin = countryOfOrigin;
        }

        public string CountryOfOrigin
        {
            get
            {
                return countryOfOrigin;
            }
            set
            {
                countryOfOrigin = value;
            }
        }

        public string Material
        {
            get
            {
                return material;
            }
            set
            {
                material = value;
            }
        }

        public int CompareTo(Clothes other)
        {
            throw new NotImplementedException();
        }

        public override string ToString()
        {
            return $"{base.ToString()}, Material:{material}, Contry:{countryOfOrigin}";
        }
    }
}